lisp/dank-comments.el

```diff
@@ -119,23 +114,43 @@ Optional STARTING-COMMENT-ID will start the comment tree at the comment (instead
                                                                 :depth dank-comments-default-depth
                                                                 :comment starting-comment-id))
          (post (dank-post-parse (car post-comments)))
-         (comments (mapcar #'dank-comment-parse (cdr post-comments))))
+         (post-author (dank-post-author post))
+         (comments (mapcar (lambda (c) (dank-comment-parse c post-author)) (cdr post-comments))))
     (setq dank-comments-current-post post
           dank-comments-current-comments comments)
     (dank-comments-set-header-line)))
 
+(defun dank-comments--set-comments-ewoc (ewoc comments)
+  "Populate the EWOC with COMMENTS."
+  (let* ((flattened-comments (flatten-tree comments)))
+    (mapc (lambda (c) (ewoc-enter-last ewoc c)) flattened-comments)
+    ewoc))
+
+(defun dank-comments--insert-comments-at-pos-ewoc (ewoc comments pos)
+  "Insert COMMENTS into EWOC at POS.
+This deletes the ewoc node at POS before inserting COMMENTS."
+  (let* ((inhibit-read-only t)
+         (node (ewoc-locate ewoc pos))
+         (delete-node node)
+         (flattened-comments (flatten-tree comments)))
+    (while (consp flattened-comments)
+      (setq node (ewoc-enter-after ewoc node (pop flattened-comments))))
+    (ewoc-delete ewoc delete-node)))
```

Comment:
@john2x FYI, I'm not sure exactly what you're doing here, but you might want to use `ewoc-invalidate` rather than deleting and adding a node.

Reply:
This is replacing the `[N more comments]` node with the new comment tree being loaded.

I was hoping to use the `refresh` methods of `ewoc`, but wasn't sure how to make it fit. Refreshing the comments list is basically clearing the list and getting a new one.

I ended up creating a new `ewoc` on refresh, banking on the GC to clean it up since there was no method to delete it. Hopefully refreshes shouldn't happen often in real life usage.

Reply:
Hm, IIUC that means recreating the whole comment tree when a subtree is expanded, right?  I'm guessing that would be fine much of the time, but in large threads, it might be noticeably slow.

I think that, if you could use nested EWOCs, that a subtree could be represented by its own EWOC, and you could `invalidate` that subtree's node, which would cause only that subtree to be recreated.  But I'm opining from the sidelines here, as I haven't tried that yet, myself.  :)  (When Matrix gains threading support, I'll probably have to dig into this for Ement.el.)

Reply:
All good :) Yeah nested EWOCs sound like a prefect fit for the comment trees. I'll have to revisit it later. I'm opting to keep it like this for now